### PR TITLE
4306 Added notification for unsupported GeoServer projections.

### PR DESCRIPTION
**Pull request by kjlippold**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW DUPLICATED LINES DENSITY | 41.5584415584416 |
| NEW CODE SMELLS | 4 |
| NEW SECURITY HOTSPOTS | 0 |
| NEW VULNERABILITIES | 0 |
| NEW BUGS | 1 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4311](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4311)

File: hs_file_types/models/geofeature.py [hunk 1]
```diff
@@ -155,6 +155,24 @@ def get_preview_data_url(self, resource, folder_path):
...
         return preview_data_url
 
+    def check_valid_geoserver_projection(self):
+        """Check if the content is using a projection supported by GeoServer."""
+
+        try:
+            proj = osr.SpatialReference(wkt=str(self.originalcoverage.projection_string))
+            authority = ':'.join((proj.GetAttrValue('AUTHORITY', 0), proj.GetAttrValue('AUTHORITY', 1),))
+
+            with open(os.path.dirname(__file__) + '/../geoserver_epsg.txt') as f:
```
[the remark this answers is not in the document]
Comment (kjlippold) [reply hs_file_types/models/geofeature.py:165]: Where would you recommend storing that type of metadata? I asked Alva about that when we first set up GeoServer and he said HydroShare didn't have a way to handle system metadata that isn't modifiable by users. That's why we went the route of generating the GeoServer URLs each time the page was loaded instead of doing it once and storing them somewhere.

[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_file_types/models/geofeature.py:165]: There is an extra_data field on each logical file for this type of thing.  @hyi - used it for storing the URL in a .url file.

https://github.com/hydroshare/hydroshare/blob/develop/hs_file_types/models/base.py#L723

File: theme/static/js/resourceLandingAjax.js [hunk 2]
```diff
@@ -800,6 +802,34 @@ function get_irods_folder_struct_ajax_submit(res_id, store_path) {
...
+                            '<span aria-hidden="true">&times;</span>' +
+                        '</button>' +
+                        '<div class="flex">' +
+                            '<i class="glyphicon glyphicon-info-sign" style="margin-right: 20px;"></i>' +
+                            '<em>' +
+                                'One or more files are using a projection not supported by GeoServer. ' + 
+                                'If you want access to GeoServer features for those files, please make sure ' + 
+                                'they are using a supported projection. ' + 
+                                '<br><strong><a ' +
+                                    'href="' + geoserver_url + '/hydroshare/wms?request=GetCapabilities" ' +
```
[the remark this answers is not in the document]
Comment (kjlippold) [reply theme/static/js/resourceLandingAjax.js]: Moved GetCapabilities URL to settings.py: https://github.com/hydroshare/hydroshare/pull/4311/commits/28ca8e9e26ceea3567f096b8545b3248b7b9cf8f

File: hs_file_types/models/geofeature.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kjlippold) [reply hs_file_types/models/geofeature.py]: I've moved the projection checks to the update_web_services task, so the check will only happen when web services are updated, and the result is stored in extra_data: https://github.com/hydroshare/hydroshare/pull/4311/commits/0c1176adc9001ccf39b2a2958b7b01e320446c6a

File: hs_file_types/models/base.py [hunk 3]
```diff
@@ -702,6 +702,12 @@ def get_preview_data_url(self, resource, folder_path):
         # - subclass needs to override this
         return None
 
+    @classmethod
```
Comment (sblack-usu) [inline hs_file_types/models/base.py:705]: This method is no longer used and should be deleted

File: hs_core/views/resource_folder_hierarchy.py [hunk 4]
```diff
@@ -169,6 +171,8 @@ def data_store_structure(request):
                                         resource=resource,
                                         folder_path=f_store_path
                                      ),
+                                     'valid_geoserver_projection':
+                                        f.logical_file.extra_data.get('valid_geoserver_proj', 'True'),
```
Comment (sblack-usu) [inline hs_core/views/resource_folder_hierarchy.py:175]: Should this default to False?

**Review by sblack-usu — CHANGES_REQUESTED** [review]
This looks great.  Just need an unused method to be deleted and I'll approve.

File: hs_file_types/models/base.py (hunk 3 above)
Comment (kjlippold) [reply hs_file_types/models/base.py]: I’ve removed this method here: https://github.com/hydroshare/hydroshare/pull/4311/commits/88b583f6fac4113fe7448ffb38aa2e57ec97714e
